webview/src/plots/hooks/useMutationObserver.ts

```diff
@@ -0,0 +1,27 @@
+import { MutableRefObject, useEffect, useRef } from 'react'
+
+const OPTIONS = { childList: true, subtree: true }
+
+export const useMutationObserver = (
+  targetEl: MutableRefObject<HTMLElement | null>,
+  onChange: () => void
+) => {
+  const observer = useRef<MutationObserver | null>(null)
+
+  useEffect(() => {
+    if (targetEl.current && !observer.current) {
+      observer.current = new MutationObserver(onChange)
```

Comment:
The only way I could think of to update the vega actions is with the help fo the `MutationObserver` API. I wonder if there is a "less hacky" way of doing this though 🤔

Reply:
IIUC the below screenshot shows that the vega-actions are already in the DOM when the modal is opened and the button has not been click on yet:

![image](https://github.com/iterative/vscode-dvc/assets/37993418/eb8e7b1b-68d6-4c52-ba22-c738d39dfc94)

Doesn't that mean that you can append the new action in the original `useEffect` in `ZoomedInPlot` no need to use the `MutationObserver`?

Reply:
The `vega-actions` do render before the button has been clicked but they don't render at the time of mounting, they render a little later after that:

```
  useEffect(() => {
    const modalOpenClass = 'modal-open'
    document.body.classList.add(modalOpenClass)

    debugger

    return () => {
      document.body.classList.remove(modalOpenClass)
    }
  }, [])
```

<img width="1332" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/15d1e3c0-6b2d-41dc-ae44-e206bfe50f4f">

I use the `MutationObserver` to detect when the `vega-actions` are actually rendered the DOM.